dowhy/gcm/distribution_change.py

```diff
@@ -88,11 +88,34 @@ def mechanism_change_test(
         return conditional_independence_test(joint_target_samples, data_set_indices, joint_parent_data)
 
 
+def remove_invariant_nodes(
+    invariant_nodes: list,
+    causal_model: ProbabilisticCausalModel,
+    old_data: pd.DataFrame,
+    auto_assignment_quality: Optional[AssignmentQuality],
+):
+    for invar_node in invariant_nodes:
+        if invar_node in list(causal_model.graph.nodes):
+            # Get parent and child nodes
+            parents = get_ordered_predecessors(causal_model.graph, invar_node)
+            children = list(causal_model.graph.successors(invar_node))
+            # Remove the middle node
+            causal_model.graph.remove_node(invar_node)
+            # Connect parent and child nodes
+            for parent in parents:
+                for child in children:
+                    causal_model.graph.add_edge(parent, child)
+            # Update the causal mechanism for the child nodes
+            for child in children:
+                set_causal_mechanism_for_node(causal_model, child, old_data, quality=auto_assignment_quality)
```

Comment:
@bloebp  can you please explain more on this....i didnt understood here!

Reply:
What I meant is something like:

```
if auto_assignment_quality is not None:
   for child in children:
      set_causal_mechanism_for_node(causal_model, child, old_data, quality=auto_assignment_quality)
```

In that way, one can still get the modified model without running the auto assignment on the children nodes by providing `auto_assignment_quality=None`.

Reply:
If we do this then we will not be able to get updated model for child nodes when assignment quality is None right? and this is which we dont want. Or i am still unable to understand the reason. Also, please see the below image where this function is getting called, its called after the assigning of the old causal graph models...

![image](https://github.com/py-why/dowhy/assets/68959880/8e80b520-b789-4644-bb55-a6ddd8a11eea)

Reply:
Ok yes, nevermind my comment, I was thinking of a use-case outside of this method where a user just wants the corrected graph structure without necessarily having a new mechanism assigned. But then we would also need to make the data optional.

Let's keep the logic as is, but to avoid confusion, the assignment quality should not be `Optional`. This means, in the `assign_causal_mechanism_node` function, set it to the default `quality: AssignmentQuality = AssignmentQuality.GOOD` in the method definition and remove the `Optional[...]` typing in `assign_causal_mechanism_node` as well as in `_remove_invariant_nodes`. It should be clearer then.